### PR TITLE
Feature/contitional test runner

### PR DESCRIPTION
**Pull request by rPraml**
Hello Rob,

this PR adds a "ConditionalTestRunner", so that you can exclude/include tests for certain platforms.
I would prefer
```java
@ForPlatform(Platform.H2)
@Test
public void myTest() {
   ...
}
```
instead of
```java
@Test
public void myTest() {
  if (!isH2()) {
    return;
  }
   ...
}
```
as this gives you a better overview about test statistics.
The first test would be counted as "skipped" if platform does not match, the second test as "passed" although it did not test anything.

### CODE REVIEW

File: src/test/java/io/ebean/PrimaryServerTest.java [hunk 1]
```diff
@@ -19,8 +20,8 @@ public void testIsSkipPrimaryServer() throws Exception {
     assertFalse(PrimaryServer.isSkip());
   }
 
-  @Ignore
   @Test
+  @ForPlatform(Platform.H2)
```
Comment (rPraml) [inline src/test/java/io/ebean/PrimaryServerTest.java]: enabled this test for H2

File: src/test/java/io/ebean/UpdateQueryTest.java [hunk 2]
```diff
@@ -30,6 +33,7 @@ public void basic() {
   }
 
   @Test
+  @IgnorePlatform(Platform.SQLSERVER)
```
Comment (rPraml) [inline src/test/java/io/ebean/UpdateQueryTest.java]: TableAlias does not work on sqlserver, only when the (slower) select mode=cursor is used. (but then other tests will fail)

File: src/test/java/io/ebean/annotation/ForPlatform.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package io.ebean.annotation;
```
Comment (rPraml) [inline src/test/java/io/ebean/annotation/ForPlatform.java]: Maybe you can take a look at this. I see that you change lots of package-names to be java9 compatible. I don't know if duplicate packages causes issues in unittests (there's no need to move this annotation to the ebean-annotations artifact IMHO)

Comment (rbygrave) [reply src/test/java/io/ebean/annotation/ForPlatform.java]: > no need to move this annotation to the ebean-annotations

I agree. LGTM

**Pull request merged**